lib/rules/selector-not-notation/index.js

```diff
@@ -0,0 +1,195 @@
+const isStandardSyntaxSelector = require('../../utils/isStandardSyntaxSelector');
+const parseSelector = require('../../utils/parseSelector');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const {
+	isPseudoClass: buggyIsPseudoClass, // cf https://github.com/postcss/postcss-selector-parser/blob/master/src/selectors/guards.js#L52
+	isAttribute,
+	isClassName,
+	isUniversal,
+	isIdentifier,
+	isTag,
+} = require('postcss-selector-parser');
+
+const ruleName = 'selector-not-notation';
+const messages = ruleMessages(ruleName, {
+	expected: (type) => `Expected ${type} :not() pseudo-class notation`,
+});
+const meta = { url: 'https://stylelint.io/user-guide/rules/list/selector-not-notation' };
+
+/** @typedef {import('postcss-selector-parser').Node} Node */
+/** @typedef {import('postcss-selector-parser').Selector} Selector */
+/** @typedef {import('postcss-selector-parser').Pseudo} Pseudo */
+/** @typedef {import('postcss-selector-parser').Container} Container */
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+const isPseudoClass = (node) =>
+	buggyIsPseudoClass(node) &&
+	node.value.toLowerCase() !== ':first-line' &&
+	node.value.toLowerCase() !== ':first-letter';
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+const isSimpleSelector = (node) =>
+	isPseudoClass(node) ||
+	isAttribute(node) ||
+	isClassName(node) ||
+	isUniversal(node) ||
+	isIdentifier(node) ||
+	isTag(node);
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+const isNot = (node) =>
+	isPseudoClass(node) && node.value !== undefined && node.value.toLowerCase() === ':not';
+
+/**
+ * @param {Selector[]} list
+ * @returns {boolean}
+ */
+const isSimple = (list) => {
+	if (list.length > 1) return false;
+
+	const [first, second] = list[0].nodes; // list is never empty
+
+	if (!first) return true;
+
+	if (second) return false;
+
+	return isSimpleSelector(first) && !isNot(first);
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary, _, context) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: ['simple', 'complex'],
+		});
+
+		if (!validOptions) return;
+
+		root.walkRules((ruleNode) => {
+			if (!isStandardSyntaxRule(ruleNode)) return;
+
+			const selector = ruleNode.selector;
+
+			if (!selector.includes(':not(')) return;
+
+			if (!isStandardSyntaxSelector(selector)) return;
+
+			const fixedSelector = parseSelector(selector, result, ruleNode, (container) => {
+				container.walkPseudos((pseudo) => {
+					if (!isNot(pseudo)) return;
+
+					if (primary === 'complex') {
+						const prev = pseudo.prev();
+						const hasConsecutiveNot = prev && isNot(prev);
+
+						if (!hasConsecutiveNot) return;
+
+						if (context.fix) return fixComplex(/** @type {Pseudo} */ (prev));
```

Comment:
```suggestion
						if (context.fix) return fixComplex(prev);
```

Reply:
After line 98 we known we have a `Pseudo` but `prev()` returns a `Node` hence the casting so that it doesn't complain.